### PR TITLE
[T3A4][T17-A3]Abdul Rahman

### PR DESCRIPTION
**Pull request by arshrahman**
Add new exception to handle the case of missing file. Ready for review!~

### CODE REVIEW

**Comment by arshrahman** [comment]
[T3A5][T17-A3]Abdul Rahman

File: src/seedu/addressbook/commands/CommandResult.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 public class CommandResult {
```
Comment (JoelT-92) [inline src/seedu/addressbook/commands/CommandResult.java]: Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 2]
```diff
@@ -82,31 +91,41 @@ private static boolean isValidPath(Path filePath) {
      * Saves all data to this storage file.
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * @throws StorageFileNotFoundException
```
Comment (JoelT-92) [inline src/seedu/addressbook/storage/StorageFile.java]: This function does not throw StorageFileNotFoundException

File: src/seedu/addressbook/Main.java [hunk 3]
```diff
@@ -107,7 +107,10 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
-            storage.save(addressBook);
+            if (!storage.save(addressBook)) {
```
Comment (JoelT-92) [inline src/seedu/addressbook/Main.java]: It is usually not recommended to catch one exception and throw another. Instead, if you take a look at the source code, you may see that the call to storage.save is wrapped within a try-catch block. Furthermore, the catch statement is set to catch all exceptions. This means that you can simply throw the exception from storage.save and let the original code handle it.

**Comment by JoelT-92** [comment]
@arshrahman Some comments added. Please close the PR after reading comments. Good job!

**Comment by arshrahman** [comment]
@JoelT-92 Thanks Joel for your valuable comments. 
Will take note of them and apply them in future learning activities. 
Made a mistake by committing to the same branch. Won't happen again. Sorry!

**Pull request closed without merging**